### PR TITLE
ci: Remove the cyclic peer-dependency (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
Since pnpm upgrade the build task is trying to compile `core` and `nodes-base` simultaneously because of the cyclic dependency. This can cause issues when core takes longer to compile than `nodes-base. This PR removes the cyclic dependency and fixes the build process.

## Review / Merge checklist
- [x] PR title and summary are descriptive

### CODE REVIEW

File: packages/core/test/helpers/constants.ts [hunk 1]
```diff
@@ -4,11 +4,11 @@ import type {
...
-import { If } from 'n8n-nodes-base/nodes/If/If.node';
-import { Merge } from 'n8n-nodes-base/dist/nodes/Merge/Merge.node';
-import { NoOp } from 'n8n-nodes-base/nodes/NoOp/NoOp.node';
-import { Set } from 'n8n-nodes-base/nodes/Set/Set.node';
-import { Start } from 'n8n-nodes-base/nodes/Start/Start.node';
+import { If } from '../../../nodes-base/dist/nodes/If/If.node';
+import { Merge } from '../../../nodes-base/dist/nodes/Merge/Merge.node';
+import { NoOp } from '../../../nodes-base/dist/nodes/NoOp/NoOp.node';
+import { Set } from '../../../nodes-base/dist/nodes/Set/Set.node';
+import { Start } from '../../../nodes-base/dist/nodes/Start/Start.node';
```
Comment (ivov) [inline packages/core/test/helpers/constants.ts:11]: To better understand - This change is only to unblock CI? As normally we wouldn't do cross-package imports like this?

**Review by despairblue — APPROVED** [review]
Do you want to test this with the MySQL workflow? That one's been failing for me consistently on CI. 

Or do you just want to merge it an try it on our PRs?

**Comment by netroy** [comment]
> Do you want to test this with the MySQL workflow?

let's just rebase our branches. I'm most certain that those builds will be fixed if the issues was just this cyclic dependency.

**Comment by janober** [comment]
Got released with [`n8n@1.45.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.45.0)

**Pull request merged**